### PR TITLE
Pie chart into stimulus controller

### PR DESCRIPTION
**Pull request by KNAthari**
### Changes
- Created a stimulus controller titled `chart_controller.js`
- Transferred pie chart code to the stimulus controller
- Installed chart.js via npm
- increased width and height of pie chart
- Pie chart renders on page

### Link to Issue 
(https://app.asana.com/0/1206510424537042/1207172120169277/f)

### Comments, Notes, or  Questions
Previously, pie chart code was held in the script tag in the index.html.erb file of the knowing_categories folder. I transferred that code to the newly created stimulus controller.

### Related Screenshots 
Screenshot 2024-05-13 at 9 06 47 AM

### Before you submit for review, Did you...

- [x] Added relevant tests (if applicable)
- [x] Linked Asana ticket to PR
- [x] Added comments, notes, or questions for reviewwer
- [x] Added relevant screenshots
- [x] Ensured ticket title is clear and concise(i.e. 'Donation Form Feature' instead of 'sfg/chels/donation-form'

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,5 +1,7 @@
 {
   "dependencies": {
+    "20": "^1.0.0",
```
Comment (edithemmings) [inline package.json:3]: ^What is "20" doing?

File: config/importmap.rb [hunk 2]
```diff
@@ -2,8 +2,11 @@
...
 # Pin npm packages by running ./bin/importmap
 
-pin 'application'
-pin '@hotwired/turbo-rails', to: 'turbo.min.js'
-pin '@hotwired/stimulus', to: 'stimulus.min.js'
-pin '@hotwired/stimulus-loading', to: 'stimulus-loading.js'
+pin 'application', preload: true
+pin '@hotwired/turbo-rails', to: 'turbo.min.js', preload: true
+pin '@hotwired/stimulus', to: 'stimulus.min.js', preload: true
+pin '@hotwired/stimulus-loading', to: 'stimulus-loading.js', preload: true
```
Comment (edithemmings) [inline config/importmap.rb:8]: ^What was the reason for adding preload?

**Review by edithemmings — COMMENTED** [review]
Looks awesome nice work! Just left a few questions about some of the additions.

File: config/importmap.rb (hunk 2 above)
Comment (KNAthari) [reply config/importmap.rb:8]: The preload was a suggestion to improve the performance of the respective assets. In other words, so, they load quicker when the page loads.

File: package.json (hunk 1 above)
Comment (KNAthari) [reply package.json:3]: When I was downloading some of the dependencies early on, I kept getting a warning suggesting that my npm version needs an update, which is represented by the 20.

Comment (edithemmings) [reply package.json:3]: This is not that. It's some other node package: https://www.npmjs.com/package/20

Please remove it. It is a security risk to have unknown packages installed on our computers

Comment (KNAthari) [reply package.json:3]: It has been removed.

**Review by edithemmings — DISMISSED** [review]
🎉🚀 Great work! Thanks for addressing the comments

**Review by edithemmings — APPROVED** [review]
👍

**Review by AynAhmed — APPROVED** [review]
Just looked at it, good job on moving the pie chart logic to the controller. Looks like there where a lot of moving parts 👍

**Pull request merged**